### DISCUSSION TITLE
add_config_value and required configuration parameters

### DISCUSSION THREAD

**Discussion by BasicWolf** [sphinx-doc/sphinx#13171] [Q&A]
Hello Sphinx community, 

I'm developing my first Sphinx extension. It seems that there is no common way to mark configuration parameters as "required", without any default values. I suspect there are solid reasons for such API design. Can anyone please explain why?

**Comment by liborjelinek** [comment]
Welcome among Sphinx extensions programmers :-)

To your question. Yes, conf.py value is are optional, because they have some defaults. Sphinx can build your docs even without conf.py.

If some value is actually required, I would e.g., raise ValueError exception in the extension.
